### DISCUSSION TITLE
Action works on one repo but fails on another

### DISCUSSION THREAD

**Discussion by noah-de** [community/community#42797] [Actions] [answered]
### Select Topic Area

Question

### Body

I cannot tell why this action would fail on our production instance but run without issues on the test instance.

I have an action to run some python scripts and then build a Hugo website.
Our production instance fails on the pipenv operation saying that python is not found.

`Run pipenv install
  pipenv install
  shell: /bin/bash -e {0}
  env:
    HUGO_VERSION: 0.79.1
    pythonLocation: /opt/hostedtoolcache/Python/[3](https://github.com/ucsb-seclab/seclab-website/actions/runs/3622601871/jobs/6107510455#step:6:3).8.1[4](https://github.com/ucsb-seclab/seclab-website/actions/runs/3622601871/jobs/6107510455#step:6:4)/x[6](https://github.com/ucsb-seclab/seclab-website/actions/runs/3622601871/jobs/6107510455#step:6:6)4
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.8.14/x64/lib
Warning: Python 3.[8](https://github.com/lab/lab-website/actions/runs/3622601871/jobs/6107510455#step:6:9).10 was not found on your system...
Neither 'pyenv' nor 'asdf' could be found to install Python.
You can specify specific versions of Python with:
$ pipenv --python path/to/python
Error: Process completed with exit code 1.`

The difference in output between the setup portions of the job that succeeds and the one  that fails appears to be the version of CPython that is setup. The successful job gets 3.8.15, while the failed run gets  3.8.14:
`
Run actions/setup-python@v2
  with:
    python-version: 3.8
    architecture: x64
    token: ***
  env:
    HUGO_VERSION: 0.79.1
Successfully setup CPython (3.8.15)`

The same action is producing 2 different results. Can anyone help me to understand how to understand the difference?
Thank you!

**Comment by airtower-luna** [comment, marked as answer]
I can't see from that snippet what exactly is going on, but I notice that the version of `actions/setup-python` you're using is severely outdated. The [current stable series is `v4`](https://github.com/actions/setup-python/releases), so the first thing I'd try is upgrading. :wink: 

If that doesn't help try linking (or at least sharing) the workflow file, any subtle difference between the instances might be relevant.

**Reply by noah-de** [reply]
Greetings!
Thank you very much for the quick reply!

Issue was resolved by doing 2 things:
  - incrementally going through each action and updating to the latest supported version
  - clearing the cache 

Thank you for the suggestions!
